src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -4,42 +4,55 @@
 
 #include "lazy_tensor.hpp"
 
-using ov::npuw::weights::ConcatMeta;
-using ov::npuw::weights::ConstPtr;
 using ov::npuw::weights::LazyTensor;
-using ov::npuw::weights::OrigData;
-using ov::npuw::weights::Transform;
-using ov::npuw::weights::TransformType;
-using ov::npuw::weights::UnpackMeta;
 
 namespace ov {
 namespace npuw {
 namespace weights {
 
+enum class TransformType : int { TENSOR, CONST, CONCAT, UNPACK, PERMUTE, CONVERT };
+using ConcatMeta = std::pair<std::vector<LazyTensor>, std::size_t>;
+using UnpackMeta = std::tuple<LazyTensor, LazyTensor, LazyTensor, ov::Shape, ov::element::Type>;
+using ConstPtr = std::shared_ptr<ov::op::v0::Constant>;
+using Transform = std::variant<ov::Tensor, ConstPtr, ConcatMeta, UnpackMeta, std::vector<std::size_t>, std::monostate>;
+
 struct LazyTensorImpl {
 public:
     LazyTensorImpl() = default;
-    LazyTensorImpl(const TransformType& type, const Transform& transform);
-
-    bool operator==(const LazyTensorImpl& other) const;
+    LazyTensorImpl(const ov::Tensor& tensor);
+    LazyTensorImpl(const std::shared_ptr<ov::op::v0::Constant>& const_ptr);
+    LazyTensorImpl(const std::vector<LazyTensor>& to_concat, const std::size_t& axis);
+    LazyTensorImpl(const LazyTensor& cw,
+                   const LazyTensor& cz,
+                   const LazyTensor& cs,
+                   const ov::Shape& shape,
+                   const ov::element::Type& type);
```

Comment:
Instead of a dozen constructors, probably, you need only one: `explicit LazyTensorImpl(Transform &&t)`.

So we can add more Transforms somewhere else and you don't need to add any more constructors here.

Reply:
Done